### PR TITLE
replace GET bodies with query params

### PR DESCRIPTION
**Pull request by ekorman**
this replaces GET bodies with query params

image

https://stackoverflow.com/a/62468403

### CODE REVIEW

File: api/valor_api/main.py [hunk 1]
```diff
@@ -387,8 +392,8 @@ def get_datasets(
 
     Parameters
     ----------
-    filters : schemas.Filter, optional
-        An optional filter to constrain results by.
+    filters : schemas.FilterQueryParams, optional
+        An optional filter to constrain results by. All of these fields should be json strings
```
Comment (ntlind) [inline api/valor_api/main.py:396]:
```suggestion
        An optional filter to constrain results by. All fields should be specified as strings in a JSON.
```

Comment (ntlind) [reply api/valor_api/main.py:396]: the two big suggestions here are a) capitalize JSON for consistency and b) add a period at the end of the sentence

File: api/valor_api/main.py [hunk 2]
```diff
@@ -592,9 +599,11 @@ def delete_dataset(
     status_code=200,
     dependencies=[Depends(token_auth_scheme)],
     tags=["Datums"],
+    description="Fetch datums. All query parameters are optional and if specified should be JSON strings",
```
Comment (ntlind) [inline api/valor_api/main.py:602]:
```suggestion
    description="Fetch datums using optional JSON strings as query parameters.",
```

Comment (ntlind) [reply api/valor_api/main.py:602]: consider this a nit

File: api/valor_api/schemas/filters.py [hunk 3]
```diff
@@ -303,3 +305,28 @@ class Filter(BaseModel):
         extra="forbid",
         protected_namespaces=("protected_",),
     )
+
+
+def field_to_tuple(field_info):
```
Comment (ntlind) [inline api/valor_api/schemas/filters.py:310]: is this function used anywhere? the name doesn't match the output to me; I'm wondering if you forgot to delete these lines

File: api/valor_api/schemas/filters.py [hunk 4]
```diff
@@ -303,3 +305,28 @@ class Filter(BaseModel):
...
+model_fields = Filter.model_fields
+model_def_dict = {kwarg: (str | None, None) for kwarg in model_fields}
+FilterQueryParams = create_model(
+    "FilterQueryParams",
+    __config__=ConfigDict(extra="forbid"),
+    **model_def_dict,  # type: ignore
+)
+
+
+def filter_query_params_to_filter(filter_query_params) -> Filter:
```
Comment (ntlind) [inline api/valor_api/schemas/filters.py:326]: can we start this name with a verb for consistency (e.g., `convert_filter_query_params_to_filter_obj`)?

File: api/valor_api/schemas/filters.py [hunk 5]
```diff
@@ -303,3 +305,28 @@ class Filter(BaseModel):
...
+    return Filter(
```
Comment (ntlind) [inline api/valor_api/schemas/filters.py:327]: can we add a docstring for consistency with other non-private functions in this file?

File: api/valor_api/schemas/filters.py (hunk 3 above)
Comment (ekorman) [reply api/valor_api/schemas/filters.py:310]: good catch-- nope deleted

File: api/valor_api/schemas/filters.py (hunk 4 above)
Comment (ekorman) [reply api/valor_api/schemas/filters.py:326]: yep, done

File: api/valor_api/main.py (hunk 1 above)
Comment (ekorman) [reply api/valor_api/main.py:396]: yep. done

File: api/valor_api/main.py (hunk 2 above)
Comment (ekorman) [reply api/valor_api/main.py:602]: yep done

File: api/valor_api/schemas/filters.py (hunk 5 above)
Comment (ekorman) [reply api/valor_api/schemas/filters.py:335]: yep, added

**Review by ntlind — APPROVED** [review]

**Pull request merged**